packages/front-end/pages/features/FeaturesDraftTable.tsx

```diff
@@ -44,26 +46,37 @@ export default function FeaturesDraftTable() {
         return;
     }
   };
-  const featuresAndRevisions = useAddComputedFields(
-    data?.featuresAndRevisions,
-    (featureAndRevision) => {
-      const createdBy = featureAndRevision?.revision
-        ?.createdBy as EventAuditUserLoggedIn | null;
-      return {
-        id: featureAndRevision?.feature?.id,
-        tags: featureAndRevision?.feature?.tags,
-        status: featureAndRevision?.revision?.status,
-        version: featureAndRevision?.revision?.version,
-        dateUpdated: featureAndRevision?.revision?.dateUpdated,
-        project: featureAndRevision?.feature?.project,
-        creator: createdBy?.name,
-        comment: featureAndRevision?.revision?.comment,
-      };
-    }
+
+  const featuresAndRevisions = data?.revisions.reduce<FeaturesAndRevisions[]>(
+    (result, revision) => {
+      const feature = features.find((f) => f.id === revision.featureId);
+      if (feature && feature?.dateCreated <= revision.dateCreated) {
+        result.push({
+          ...revision,
+          feature,
+        });
+      }
+      return result;
+    },
+    []
   );
```

Comment:
Move this inside of the `useAddComputedFields` function instead.  Will be much cleaner and memoized correctly.

```ts
const revisions = useAddComputedFields(data?.revisions, (revision) => {
  const feature = features.find(f => f.id === revision.featureId);
  return {
     // ...
  }
}, [features]);
```

Also, notice I'm also passing in `features` as a dependency to this hook